src/Dialect/ONNX/ONNXOps/Canonicalize.cpp

```diff
@@ -584,6 +585,66 @@ class PropagateReshapeThroughBinaryOpPattern
   };
 };
 
+// This pattern bubbles up AddOp through transpose to keep the bias Add
+// operation right after LN_type op. This will helps the other patterns fold the
+// add into the operands of LayerNorm.
+//
+// From: LayerNorm
+//    |
+// Transpose
+//    |
+//   Add
+//
+// To:
+// LayerNorm
+//    |
+//   Add
+//    |
+// Transpose
+template <typename LN_TYPE>
+class BubbleUpBiasForLayerNormPattern : public OpRewritePattern<ONNXAddOp> {
```

Comment:
we can also do for RMSNorm as you are already doing
```suggestion
class BubbleUpBiasForNormOpPattern : public OpRewritePattern<ONNXAddOp> {
```